src/Widgets/DownloadsButton.vala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright © 2019 Adrian Cochrane (https://adrian.geek.nz)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Adrian Cochrane <alcinnz@lavabit.com>
+ */
+namespace Ephemeral {
+    public class DownloadsButton : Gtk.Revealer {
+        Gtk.ListBox downloads_list;
+
+        public DownloadsButton() {
+            Object ();
```

Comment:
While we're here, give the revealer a transition type that looks nice and doesn't reserve space

```suggestion
            Object (
                transition_type: Gtk.RevealerTransitionType.SLIDE_LEFT
            );
```